src/handle.c

```diff
@@ -16,10 +16,29 @@
  */
 #include "luv.h"
 
+static void* luv_newuserdata(lua_State* L, size_t sz) {
+  void* handle = malloc(sz);
+  if (handle) {
+    *(void**)lua_newuserdata(L, sizeof(void*)) = handle;
+  }
+  return handle;
+}
+
+static void* luv_checkudata(lua_State* L, int ud, const char* tname) {
+  return *(void**) luaL_checkudata(L, ud, tname);
+}
+
+static int luv_handle_gc(lua_State* L) {
+  void** udata = lua_touserdata(L, 1);
+  free(*udata);
```

Comment:
This is a problem if the userdata is referenced by the uv_loop_t structure, you need to ensure the handle is closed with `uv_close`, but even so you can't free it immediately without running the loop at least once.